src/bin/train.cc

```diff
@@ -183,11 +184,12 @@ void initializeLogger(
 nmtkit::Vocabulary * createVocabulary(
```

Comment:
I think basically one parameter should have only one meaning to prevent abusing their values as multiple meanings. In this case, CharacterVocabulary and WordVocabulary could take more 1 parameter to choose unk filtering strategy (just specified in config file) to prevent increasing tne number of meanings in unk_frequency.